### ISSUE TITLE
Project Design Principles

### ISSUE THREAD

**Issue by Madeline-UX** [zarf-dev/zarf#583]
**Is your feature request related to a problem? Please describe.**
A clear set of Zarf design principless that are used to inform the creation and evolution of the Zarf product

**Describe the solution you'd like**
I would like to establish design principles to guide future iterations of the product (ex. Keep users informed of system status with constant feedback.) To be posted on our docs site for transparency with our community.

**Additional context**
References to articles regarding design principles if you need more info.

https://www.interaction-design.org/literature/topics/design-principles

https://megdouglashowie.medium.com/os-design-9dfb35f319de

**Add thoughts or recommendations below**

**Comment by Madeline-UX** [comment]
Here are a few design principles I propose:
- Agnostic Tech Stack: Make tool compatible with different system requirements or tech stacks (ex. OS, Distro, etc)
- Customizable: so novices and experts can choose to do more or less 
- Collaborative design: Include developers and users in design process
- Constant Feedback: Keep users informed of system status with clear and timely feedback
- Minimalist Design: Don’t clutter with unnecessary items.
- Prevent Errors: When possible inform users of the outcomes of an action or reduce user errors (ex. select option vs type out command)

**Comment by Madeline-UX** [comment]
Some interesting insights on design principles

https://uxdesign.cc/the-design-principle-paradox-4ef2686c2cb6

https://uxdesign.cc/design-principles-are-dead-634eee4a6ded

**Comment by brandtkeller** [comment]
I agree with the need for these - taking these and the evolving need over to #3552
